src/main/java/us/kilroyrobotics/subsystems/Tower.java

```diff
@@ -0,0 +1,288 @@
+
+                    currentLevel = 4;
+                    setState(TowerState.POINT_WHEELS_BACK);
+                } else if ((isTriggered(TowerEvent.GOTO_L3)) && (currentLevel != 3)) {
+                    elevator.setPosition(ElevatorConstants.kL3Height);
+                    wrist.setAngle(CoralMechanismConstants.kScoringL3);
+
+                    currentLevel = 3;
+                    setState(TowerState.RAISING_TO_L3);
+                } else if ((isTriggered(TowerEvent.GOTO_L2)) && (currentLevel != 2)) {
+                    elevator.setPosition(ElevatorConstants.kL2Height);
+                    wrist.setAngle(CoralMechanismConstants.kScoringL2);
+
+                    currentLevel = 2;
+                    setState(TowerState.RAISING_TO_L2);
+                } else if ((isTriggered(TowerEvent.GOTO_L1)) && (currentLevel != 1)) {
+                    elevator.setPosition(ElevatorConstants.kL1Height);
+                    wrist.setAngle(CoralMechanismConstants.kScoringL1);
+
+                    currentLevel = 1;
+                    setState(TowerState.RAISING_TO_L1);
+                }
+                break;
+            case POINT_WHEELS_BACK:
+                CommandScheduler.getInstance()
+                        .schedule(
+                                Commands.runOnce(
+                                        () ->
+                                                drivetrain.setControl(
+                                                        point.withModuleDirection(
+                                                                Rotation2d.kZero))));
+                setState(TowerState.BACKUP_BEFORE_SCORE);
+                break;
+            case BACKUP_BEFORE_SCORE:
+                CommandScheduler.getInstance().schedule(backupCommand);
+                setState(TowerState.WAIT_FOR_BACKUP);
+                break;
+            case WAIT_FOR_BACKUP:
+                if (backupCommand.isFinished()) setState(TowerState.RAISING_TO_L4);
+                break;
+            case SCORING:
+                if (!coralIntakeMotor.isCoralDetected() || stateTimer.hasElapsed(0.3)) {
+                    leds.setMode(LEDMode.Off);
+                    setState(TowerState.INIT);
+                }
+                break;
+        }
+    }
+
+    public void triggerEvent(TowerEvent event) {
+        pendingEvent = event;
+    }
+
+    public TowerEvent getPendingEvent() {
+        return pendingEvent;
+    }
+
+    public TowerState getState() {
+        return currentState;
+    }
+
+    private Boolean isTriggered(TowerEvent event) {
+        if (pendingEvent == event) {
+            pendingEvent = TowerEvent.NONE;
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public void setState(TowerState newState) {
+        currentState = newState;
+        stateTimer.reset();
+        System.out.println("P: " + pendingEvent + " S: " + currentState);
```

Comment:
Remove prints

Reply:
Will do before dcmp. Just keeping it there while this is still being developed.